script.py

```diff
@@ -100,24 +118,36 @@ def cleanup_backups(bucket):
     four_days_ago = datetime.now() - timedelta(days=4)
     four_days_ago = four_days_ago.date()
     try:
-        for i in range(1, max_backups):
-            folder = s3.list_objects_v2(Bucket=bucket, Prefix=f"{four_days_ago}_{i}")
-            if "Contents" in folder:
+        i = 1
+        while "Contents" in s3.list_objects_v2(
```

Comment:
you are calling `s3.list_objects_v2` 2 times in each iteration
this will slow down the code
optimize